packages/xo-web/src/xo-app/logs/index.js

```diff
@@ -153,21 +151,30 @@ class Log extends BaseComponent {
     (logId, runId) => logId !== runId
   )
 
-  _getFilteredCalls = createFilter(
+  _getCallsByState = createSelector(
     () => this.props.log.calls,
-    createSelector(
-      () => this.state.filter.value,
-      this._getIsJobInterrupted,
-      (value, isInterrupted) => PREDICATES[value](isInterrupted)
-    )
+    this._getIsJobInterrupted,
+    (calls, isInterrupted) => {
+      const callsByState = {}
+      forEach(CALL_FILTER_OPTIONS, ({ value }) => {
+        callsByState[value] = filter(calls, PREDICATES[value](isInterrupted))
+      })
+      return callsByState
+    }
+  )
+
+  _getFilteredCalls = createSelector(
+    () => this.state.filter.value,
+    this._getCallsByState,
+    (value, calls) => calls[value]
   )
 
-  _filterValueRenderer = createSelector(
-    () => this._getFilteredCalls().length,
-    ({ label }) => label,
-    (size, label) => (
+  _filterOptionRenderer = createSelector(
+    option => option,
+    this._getCallsByState,
+    ({ label, value }, calls) => (
       <span>
-        {_(label)} ({size})
+        {_(label)} ({calls[value].length})
```

Comment:
The `option` will change everytime `Select` calls this selector. So the selector's cache becomes useless. The result of the selector should actually be the renderer.
```js
_getFilterOptionRenderer = createSelector(
  this._getCallsByState,
  callsByState => option => <span>...</span>
```